tests/Aspire.Hosting.Valkey.Tests/ValkeyFunctionalTests.cs

```diff
@@ -0,0 +1,197 @@
+                    });
+
+                    hb.AddRedisClient(valkey1.Resource.Name);
+
+                    using (var host = hb.Build())
+                    {
+                        await host.StartAsync();
+
+                        var redisClient = host.Services.GetRequiredService<IConnectionMultiplexer>();
+
+                        var db = redisClient.GetDatabase();
+
+                        await db.StringSetAsync("key", "value");
+
+                        // Force Redis to save the keys (snapshotting)
+                        // c.f. https://redis.io/docs/latest/operate/oss_and_stack/management/persistence/
+
+                        await redisClient.GetServers().First().SaveAsync(SaveType.BackgroundSave);
+                    }
+                }
+                finally
+                {
+                    // Stops the container, or the Volume/mount would still be in use
+                    await app.StopAsync();
+                }
+            }
+
+            var builder2 = CreateDistributedApplicationBuilder();
+            var valkey2 = builder2.AddValkey("valkey");
+
+            if (useVolume)
+            {
+                valkey2.WithDataVolume(volumeName);
+            }
+            else
+            {
+                valkey2.WithDataBindMount(bindMountPath!);
+            }
+
+            using (var app = builder2.Build())
+            {
+                await app.StartAsync();
+                try
+                {
+                    var hb = Host.CreateApplicationBuilder();
+
+                    hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+                    {
+                        [$"ConnectionStrings:{valkey2.Resource.Name}"] = await valkey2.Resource.ConnectionStringExpression.GetValueAsync(default)
+                    });
+
+                    hb.AddRedisClient(valkey2.Resource.Name);
+
+                    using (var host = hb.Build())
+                    {
+                        await host.StartAsync();
+
+                        var redisClient = host.Services.GetRequiredService<IConnectionMultiplexer>();
+
+                        var db = redisClient.GetDatabase();
+
+                        var value = await db.StringGetAsync("key");
+
+                        Assert.Equal("value", value);
+                    }
+                }
+                finally
+                {
+                    // Stops the container, or the Volume/mount would still be in use
+                    await app.StopAsync();
+                }
+            }
+        }
+        finally
+        {
+            if (volumeName is not null)
+            {
+                DockerUtils.AttemptDeleteDockerVolume(volumeName);
+            }
+
+            if (bindMountPath is not null)
+            {
+                try
+                {
+                    File.Delete(bindMountPath);
```

Comment:
Should this be doing `Directory.Delete`?

Reply:
Good call. Looks like we made this mistake in a few places. Fixed them all.